### ISSUE TITLE
Issues related to Accounting Conventions / Practices

### ISSUE THREAD

**Issue by RamkrishnaMondal** [prashants/webzash#29]
1. Neither any A/C nor any report of WEBZASH shows the 'Currency symbol'.... it should show as, for example 'Amount($)' / 'Amount(Rs.)' .

2. A 'Balance Sheet' is only a financial statement at a certain date, it is not a Ledger A/C , so it should not contain (Dr.) or (Cr.) symbols. 

3. The first part of webzash "Profit & Loss Statement " is actually a Trading A/C  and the second part is a real Profit & Loss A/C........ it would be better, if you rename it as "Trading and Profit & Loss A/C"

4. In Accountancy Carried Forward  and Brought Forward are used to carry balances to the next financial year ( or next accounting period).... in a combined Trading and Profit & Loss A/C,  Carried Down and Brought Down are used instead.

5. Gross Profit / Gross Loss is actually the closing balance of Trading A/C ( also called Trading Profit / Trading Loss). It is also the opening balance of Profit & Loss A/C..... so, it should be posted before posting any Indirect expenses / Indirect incomes.

**Comment by prashants** [comment]
Fixed 3, 4

**Comment by prashants** [comment]
Fixed 1

**Comment by prashants** [comment]
Need more clarification regarding 2

I am using Dr / Cr in the following places.
1. Chart of accounts
2. List of all Entries
3. Individual entry
4. Reports : Balance sheet, P&L, Trial balance, Ledger St, Ledger entries, Recon.

From the above where using Dr / Cr is valid and where its not valid ?

**Comment by RamkrishnaMondal** [comment]
Thanks for fixing 1,3 & 4.

Clarification regarding 2 : 

1. Chart of accounts.... Valid.
2. List of all Entries..... Valid.
3. Individual entry....... Valid.
4. Reports:
                  A. Balance Sheet..... Invalid, because it is not a part of 'double entry system/ Book Keeping'.  It is prepared after the closing of "books of accounts". It's main purpose is to determine the valuation of the firm / business.
   
                B. P&L : Trading and Profit & Loss A/C / Profit And Loss A/C  is a part of 'double entry system / book keeping'  so it should show Dr & Cr symbols.
                             But, P&L Statement is not a part of 'books of accounts'. it's a summery of detailed Trading and Profit & Loss A/C. It is prepared to show the earning and managing efficiency of the business to Share holders, Financers etc. It should not contain Dr / Cr symbols.

                C. Trial balance: Trial balance is prepared to make sure that the 'Sum of Dr balances' is equal to 'sum of Cr balances'. and if not equal, to find out the cause/ error and rectify that. However it has some limitations,errors such as, Amount received from Mr. Bill Gates but wrongly credited to Mr. Bill Clinton's A/C or  Paid $100 as salary but posted as $10, or The book keeper forgets to post an entry in both Dr and Cr side etc will not reflect in Trial Balance .
 It should contain Dr and Cr columns.
                  

                 D. Ledger Statement and Ledger Entry should contain Dr /Cr symbols to show which ledger A/C has what type of balance Dr/ Cr ?  

                E. BR Statement:  In theory, the  A/C balance in Passbook and In the Books of Accounts should be same, but in reality it often differs, mainly due to communication gap between Bank and the Book Keeper. 

For example,  when a cheque is deposited for collection, the Balance in Books of Accounts increases but passbook balance increases only after the amount is collected by the bank or when Bank debits bank charges / credits interests, the passbook balance changes but the same is posted in Books of Accounts only after receiving the Bank statement .   It should be prepared as follows: 


Balance as per 'Passbook' (or 'Books of Accounts')        $ XXXX
Adjustments:                                                                    $XXXX
Adjusted Balance                                                             $xxxxx  [this should be equal to Balance as per 'Books of Accounts' (or 'Passbook)' ]
   
It should not contain Dr or Cr symbols.
               
                                                                    
Hope this helps.

**Comment by prashants** [comment]
Thanks for the explanations. I will go through them.

One thing that I am stuck at is how to handle negative balance. For example a ledger under asset should be Dr balance, if in case it turns out to have Cr balance should I show it as negative number in balance sheet.

**Comment by RamkrishnaMondal** [comment]
Lets consider, we run a business where the Customers pay in advance and we supply the goods after 7 days. When we receive the advance payments, the book keeper enters as follows:

Dr     Cash / Bank A/C  $xxx
Cr     Customers A/C    $xxx

In a normal business, Customers A/C (which is  called Debtors A/C as a whole) remains Dr balance due to credit sales. But in our business it will show as Cr balance, and it's a liability for our business. we have to show it under liability in the Balance Sheet . 

General rule : Any Asset with a Cr balance is a Liability and any Liability with a Dr balance (for example, Tax paid in advance to Govt.)  is an Asset.

**Comment by prashants** [comment]
ok

**Comment by RamkrishnaMondal** [comment]
1. How Webzash should handle Current Assets & Liabilities: 
Merge 'Current Assets' and 'Current Liabilities' groups into a single group called "Current Assets & Liablities" and tell webzash that, If a particular A/C has a Dr. balance it will come under 'Current assets' in the Balance Sheet and if it's a Cr. balance it will come under "Current Liabilities' in the Balance Sheet. This is the easiest solution I can think of. 

2. Please prevent the 'Group totals amounts' to appear in the B/Sheet and P&L reports, and also prevent Dr./Cr. symbols to appear just beside each and every amount , they are making the reports over populated and ugly. Presentation does matter a lot in accountancy.

3. I can't find out any significance of opening and closing balance columns in the Trial Balance, better remove them.

4. Balance Sheet is always prepared on latest available balances of assets and liabilities ... then why choose an opening date ?  it is irrelevant .

**Comment by prashants** [comment]
No 1 will be hard to change since if its a Cr. balance in Assets I would let user change or correct it accordingly. I don't want investments appearing in current liabilities automatically.

No 2 some users might want it, though I am planning to remove Dr Cr symbols

No 3 some users might want it, some user dont. So better to keep it for now.

No 4 Since its already there, no point in removing something if its there.

No 5 Yes that can be done

**Comment by chacha** [comment]
>I don't want investments appearing in current liabilities automatically.

@prashants Why would an investment appear under liabilities? An investment is normally considered an asset with a debit balance. The only time an investment would have a credit balance is if someone invested in you, in which case it is a liability.

**Comment by chacha** [comment]
> No 4 Since its already there, no point in removing something if its there.

In order to bring it in line with what most accountants expect, I would suggest removing the opening date. It doesn't make a lot of sense and can confuse users.

**Comment by chacha** [comment]
> When there is only one Account in the database, is not it logical that a/c should be auto activated by default ?

I would greatly appreciate if the software remembered the last activated account and simply used that unless you change the account manually. Otherwise it gets very frustrating. I'll open a separate ticket for this.

**Comment by kjcpa** [comment]
For Balance Sheet, Profit & Loss Account (Net Profit) is part of Owners Equity.
"Profit & Loss Account (Net Loss)" is also part of Owners Equity. It is not an Asset item. The Net Loss will reduce the Owners Equity.

**Comment by kjcpa** [comment]
The $startdate should not affect the Balance Sheet figures if the $enddate is the same. 
Profit and Loss report is for a period, but Balance Sheet is only for a point, the end of the period.

These lines from ReportsController.php, under BALANCESHEET CALCULATIONS, may be deleted: 
$liabilities->start_date = $startdate;
$assets->start_date = $startdate;
$income->start_date = $startdate;
$expense->start_date = $startdate;
